File: benchmark.py
import numpy as np
import time

N = 1_000_000_000

start = time.perf_counter()

a = np.arange(N, dtype=np.uint64)
result = ((a * 17) ^ (a >> 3)).sum(dtype=np.uint64)

end = time.perf_counter()

print("Python:\n")
print("sum =", int(result))
print("time =", end - start)

File: benchmark.c
#include <stdio.h>
#include <stdint.h>
#include <time.h>

volatile uint64_t sink;

int main(void) {
    const uint64_t N = 1000000000ULL;
    uint64_t sum = 0;

    clock_t start = clock();

    for (uint64_t i = 0; i < N; ++i) {
        sum += (i * 17) ^ (i >> 3);
    }

    clock_t end = clock();

    sink = sum;

    printf("C:\n");
    printf("sum = %llu\n", (unsigned long long)sum);
    printf("time = %.6f s\n",
           (double)(end - start) / CLOCKS_PER_SEC);

    return 0;
}